src/middle/Mir_utils.ml

```diff
@@ -96,3 +96,68 @@ let string_of_operators =
     ; ("(OperatorAssign Divide)", "assign_divide")
     ; ("(OperatorAssign EltTimes)", "assign_elt_times")
     ; ("(OperatorAssign EltDivide)", "assign_elt_divide") ]
+
+(** [iter_stmt_transform transform stmts] will run the transform function
+    on the list of statements until the list of statements stops changing.
+
+    In our use case this could be replaced with a bottom-up map.
+*)
+let iter_stmt_transform f stmts =
+  let old_stmts = ref stmts in
+  let new_stmts = ref (f stmts) in
+  (* The count here just enforces a hard limit on the number of iterations.
+     In principle, we should only use this function with monotonic update
+     functions, but in practice we may get that wrong from time to time and
+     this count keeps us from infinite loops.
+  *)
+  let count = ref 0 in
+  while Stdlib.( != ) !old_stmts !new_stmts && !count < 100 do
+    old_stmts := !new_stmts ;
+    new_stmts := f !old_stmts ;
+    count := !count + 1
+  done ;
+  !new_stmts
```

Comment:
What are we doing with this fix-point function? The comment suggests that it is being used to simulate a bottom up transform over statement but I can't reason about how that works. 

I'm slightly uncomfortable in applying a bounded fixed point iteration (which may not converge) to what should presumably be a deterministic transform of the MIR. Is there a reason we can't just do a bottom up transform?

Reply:
Maybe we can now with the refactor! I'll try to use that functionality that you added.

Reply:
It worked!